commonsbot/state.py

```diff
@@ -68,7 +68,11 @@ def load_discussion_info(self, site):
             page = pywikibot.Page(site, 'File:' + self.file_name)
         else:
             page = self.discussion_page
-        text = page.get(get_redirect=True)
+
+        try:
+            text = page.get(get_redirect=True)
+        except pywikibot.exceptions.NoPageError:
+            text = ''
```

Comment:
Using `text = page.text` will handle the exception for you.

Reply:
Thanks for the review! :)
So you are recommending `text = page.text` instead of this whole try/except block? Will that ignore redirects as the `Page.get(get_redirect=True)` does, too?

Reply:
I see from the [docs](https://doc.wikimedia.org/pywikibot/master/api_ref/pywikibot.page.html?highlight=page#pywikibot.page.BasePage.text) the `text` property indeed loads the wikitext if necessary. You know I'm not a Python whiz, but it seems weird accessing a property would _do_ anything other than return the value!

Reply:
Yes and [yes](https://github.com/wikimedia/pywikibot/blob/398778e7d0d92221433d2ce8487e97d4e5d6e526/pywikibot/page/__init__.py#L558-L572).